src/js/home.js

```diff
@@ -0,0 +1,97 @@
+
+Handlebars.registerHelper('inc', function (index, options) {
+    return parseInt(index) + 1;
+});
+
+let slideIndex = 1;
+function plusSlides(n) {
+    showSlides(slideIndex += n);
+}
+
+function currentSlide(n) {
+    showSlides(slideIndex = n);
+}
+
+function showSlides(n) {
+    let i;
+    let slides = document.getElementsByClassName("mySlides");
+    let dots = document.getElementsByClassName("dot");
+    if (n > slides.length) { slideIndex = 1 }
+    if (n < 1) { slideIndex = slides.length }
+    for (i = 0; i < slides.length; i++) {
+        slides[i].style.display = "none";
+    }
+    for (i = 0; i < dots.length; i++) {
+        dots[i].className = dots[i].className.replace(" active", "");
+    }
+    slides[slideIndex - 1].style.display = "block";
+    dots[slideIndex - 1].className += " active";
+}
+
+
+let categoriesXHttp = new XMLHttpRequest();
+categoriesXHttp.open("GET", "categories", true);
+categoriesXHttp.onload = function () {
+    if (categoriesXHttp.status >= 200 && categoriesXHttp.status < 400) {
+        let data = JSON.parse(categoriesXHttp.responseText);
+        createCategoryContainer(data);
+    } else {
+        console.log("We conected to the server, but it returned an error.");
```

Comment:
Error handling should either be shown to user or should redirect to an error page in case of fatal error. Console logs are not preffered.